when send 0 byte data in function cl_sm_conn_out_buff_flush, the subscriber connection will close and then can not get data form server

when send 0 byte data in function cl_sm_conn_out_buff_flush, the subscriber connection will close and then can not get data form server

The client log:
2015-01-02 18:28:54 815570 [DEBUG] sysrepo_data_repository.cpp[89]:[140128593979136] (cl_sm_conn_out_buff_flush:311) Sending 207 bytes of data.
2015-01-02 18:28:54 815595 [DEBUG] sysrepo_data_repository.cpp[89]:[140128593979136] (cl_sm_conn_out_buff_flush:317) 207 bytes of data sent.
**2015-01-02 18:28:54 815651 [DEBUG] sysrepo_data_repository.cpp[89]:[140128593979136] (cl_sm_conn_write_cb:1190) fd 17 writeable
2015-01-02 18:28:54 815674 [DEBUG] sysrepo_data_repository.cpp[89]:[140128593979136] (cl_sm_conn_out_buff_flush:311) Sending 0 bytes of data.
2015-01-02 18:28:54 815763 [ERROR] sysrepo_data_repository.cpp[80]:[140128593979136] (cl_sm_conn_out_buff_flush:334) Error by writing data to fd 17: Success.
2015-01-02 18:28:54 815800 [DEBUG] sysrepo_data_repository.cpp[89]:[140128593979136] (cl_sm_connection_cleanup:203) Closing subscriber connection on fd=17.**



The server log:
2015-01-02 18:28:56 722185 [DBG] [140660327659264] (sr_cbuff_enqueue:516) Circular buffer enqueue to position=2, current count=1.
2015-01-02 18:28:56 722185 [DBG] [140660327659264] (rp_msg_process:4177) Threads: active=2/4, 1 requests in queue
2015-01-02 18:28:56 722185 [DBG] [140660327659264] (sr_cbuff_dequeue:532) Circular buffer dequeue, new buffer head=3, count=0.
2015-01-02 18:28:56 722185 [DBG] [140660327659264] (rp_get_items_req_process:793) Processing get_items request.
2015-01-02 18:28:56 722224 [INF] [140660327659264] (rp_dt_get_values_wrapper_with_opts:1351) Get items request running datastore, xpath: /ostack:ostack//., offset: 0, limit: 100
2015-01-02 18:28:56 722224 [DBG] [140660327659264] (rp_dt_prepare_data:1251) Session id = 194006278 data loaded, continue processing
2015-01-02 18:28:56 722224 [DBG] [140660327659264] (dm_get_data_info_internal:2663) Module ostack already loaded
2015-01-02 18:28:56 722224 [DBG] [140660327659264] (rp_dt_find_nodes_with_opts:113) Get_nodes opts with args: /ME:ME//. 100 0
2015-01-02 18:28:56 726349 [DBG] [140660310873856] (pm_save_data_tree:179) Persist data tree successfully saved.
2015-01-02 18:28:56 726374 [DBG] [140660310873856] (sr_locking_set_unlock_close_fd:858) File /ostack/etc/sysrepo/data/ostack.persist (fd = 13) has been unlocked
**2015-01-02 18:28:56 726416 [DBG] [140660310873856] (pm_remove_subscriptions_for_destination:1279) Subscription entries for destination '/tmp/sysrepo-subscriptions/ostack/92.qrwqYw.sock' successfully removed from 'ostack' persist file.
2015-01-02 18:28:56 726416 [DBG] [140660310873856] (np_unsubscribe_destination:1161) Disabling running datastore fo module 'ostack'.**

--- a/src/clientlib/cl_subscription_manager.c
+++ b/src/clientlib/cl_subscription_manager.c
@@ -398,6 +398,10 @@ cl_sm_conn_out_buff_flush(cl_sm_ctx_t *sm_ctx, cl_sm_conn_ctx_t *conn)
     buff_size = buff->pos;
     buff_pos = conn->out_buff.start;
 
+    if (buff_size - buff_pos == 0) {
+        return rc;
+    }
+
     SR_LOG_DBG("Sending %zu bytes of data.", (buff_size - buff_pos));
 
     do {

